Allow asymmetric PDF export resolution

tzjtan opened the pull request:
Allow PDF exporting resolution to be specified as a tuple too, to get asymmetric DPIs sometimes used by sticker/inkjet printers, e.g 600x1200dpi.

radarhere commented:
Hi. Out of curiosity, why did you close this?

tzjtan commented:
Sorry, this is my first time sending a pull request and many things are confusing to me.
Let me learn a bit more git and retry this properly without the failed tests.

radarhere commented:
Okay, sure. Well, your tests were only failing on style points. So
```python
if not isinstance(resolution,(list,tuple)):
        resolution2D = (resolution,resolution)
```

needed a space after the commas -

```python
if not isinstance(resolution, (list, tuple)):
        resolution2D = (resolution, resolution)
```

I would have also suggested that you talk about width and height resolution, instead of 1D and 2D, unless that's an established convention. The only other things would have been providing a link in the discussion to some documentation showing the need for these different resolutions, and adding a test to https://github.com/python-pillow/Pillow/blob/master/Tests/test_file_pdf.py.

tzjtan commented:
Ok thanks, I have spotted the styling errors and got tests to pass on Travis CI.
I have also added some tests to test_file_pdf.py as you suggested, and modified the documentation at docs/handbook/image-file-formats.rst.

Please let me know if some things should be done another way in my modified pull request.

radarhere commented:
This is being attempted again in PR #6961
